yt_dlp/extractor/go.py

```diff
@@ -242,22 +251,24 @@ def _real_extract(self, url):
                     'video_type': video_type,
                     'brand': brand,
                     'device': '001',
+                    'app_name': 'webplayer-abc',
                 }
                 if video_data.get('accesslevel') == '1':
                     requestor_id = site_info.get('requestor_id', 'DisneyChannels')
+                    software_statement = site_info.get('software_statement')
                     resource = site_info.get('resource_id') or self._get_mvpd_resource(
-                        requestor_id, title, video_id, None)
+                        site_info.get('resource_channel'), title, video_id, None)
                     auth = self._extract_mvpd_auth(
-                        url, video_id, requestor_id, resource)
+                        url, video_id, requestor_id, resource, software_statement)
                     data.update({
                         'token': auth,
                         'token_type': 'ap',
-                        'adobe_requestor_id': requestor_id,
+                        'adobe_requestor_id': site_info.get('resource_channel'),
```

Comment:
surely this can't be null ?
```suggestion
                        'adobe_requestor_id': site_info.get('resource_channel') or requestor_id,
```